File: BudgetThis2017/wwwroot/js/settingsController.js
(function () {
  "use strict";
  angular.module("app-budgetthis")
    .controller("settingsController", settingsController);

  function settingsController($routeParams, $http) {
    var vm = this;
    vm.errorMessage = "";
    vm.settings = {};

    $http.get("/api/settings/")
      .then(function (response) {
        angular.copy(response.data, vm.settings);
      },
      function (error) {
        vm.errorMessage = "failed to load settings.";
      })
      .finally(function () {
        vm.isBusy = false;
      });

    vm.saveSettings = function () {
      vm.isBusy = true;
      vm.message = "";
      vm.errorMessage = "";
      $http.put("/api/settings/update/" + vm.settings.userId, vm.settings)
        .then(function (response) {
          vm.message = "Settings Saved.";
        }, function (error) {
          vm.errorMessage = "failed to update settings.";
        })
        .finally(function () {
          vm.isBusy = false;
        });
    };
  }
}
)();

File: BudgetThis2017/Controllers/api/SettingsController.cs
using AutoMapper;
using BudgetThis2017.Models;
using BudgetThis2017.ViewModels;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Identity;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using System;
using System.Threading.Tasks;

namespace BudgetThis2017.Controllers.api
{
  public class SettingsController : Controller
  {
    private UserManager<BudgetUser> _userManager;
    private IBudgetThisRepository _repository;
    private ILogger<SettingsController> _logger;

    public SettingsController(IBudgetThisRepository repository, ILogger<SettingsController> logger, UserManager<BudgetUser> userManager)
    {
      _repository = repository;
      _logger = logger;
      _userManager = userManager;
    }


    [HttpGet("api/settings")]
    public IActionResult Get()
    {
      try
      {
        var results = _repository.GetSettingsByUser(new Guid(_userManager.GetUserId(HttpContext.User)));
        return Ok(Mapper.Map<SettingsViewModel>(results));
      }
      catch (Exception ex)
      {
        _logger.LogError("Error: " + ex.Message);
        return BadRequest("Error happened: " + ex.Message);
      }
    }

    //update
    [HttpPut("api/settings/update/{id}")]
    public async Task<IActionResult> Put([FromBody]SettingsViewModel settings)
    {
      if (ModelState.IsValid)
      {
        var thisSetting = Mapper.Map<Setting>(settings);
        thisSetting.UserID = new Guid(_userManager.GetUserId(HttpContext.User));
        _repository.UpdateSettings(thisSetting);

        if (await _repository.SaveChangesAsync())
        {
          return Created($"api/settings/{thisSetting.UserID}", Mapper.Map<SettingsViewModel>(thisSetting));
        }
      }
      return BadRequest("Failed to save the Settings");
    }
  }
}
